src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -134,10 +144,17 @@ def ramp_fit(model, buffsize, save_opt, readnoise_2d, gain_2d,
         Object containing optional GLS-specific ramp fitting data for the
         exposure
     """
+    if suppress_one_group:
+        log.info("Ramps with one good group suppressed as no good group.")
+        if model.data.shape[1] == 1:
+            # Suppressing one ramp groups and having data with only one group
+            # means there is no work to do.
```

Comment:
Actually, I just noticed that in the Cal WG request for this change they also said that they did NOT want this to happen when NGROUPS=1. I forgot to put that into the description of the implementation ticket. So there needs to be a little additional logic, such that when NGROUPS=1 (`model.data.shape[1] = 1`) this new switch gets ignored and slopes are still calculated from the 1 available group (at least wherever that 1 group is good). So it's only when NGROUPS>1 that we want this switch to have an effect.